### PR TITLE
Remove tools.json

### PR DESCRIPTION
**Pull request by tcdent**
This addresses #57

### CODE REVIEW

File: agentstack/generation/tool_generation.py [hunk 1]
```diff
@@ -11,131 +13,144 @@
...
 TOOL_INIT_FILENAME = "src/tools/__init__.py"
 AGENTSTACK_JSON_FILENAME = "agentstack.json"
-
+FRAMEWORK_FILENAMES: dict[str, str] = {
+    'crewai': 'src/crew.py', 
+}
+
+def get_framework_filename(framework: str, path: str = ''):
+    try:
+        return FRAMEWORK_FILENAMES[framework]
```
Comment (bboynton97) [inline agentstack/generation/tool_generation.py:22]: love

**Comment by bboynton97** [comment]
we also need to update the logic for `agentstack tools list`

**Comment by tcdent** [comment]
Just took care of #26 as well. 

I'll stop throwing stuff into this PR now 😇

File: agentstack/generation/tool_generation.py [hunk 2]
```diff
@@ -1,144 +1,182 @@
...
+    shutil.copy(tool_file_path, f'{path}src/tools/{tool_name}_tool.py')  # Move tool from package to project
+    add_tool_to_tools_init(tool_data, path)  # Export tool from tools dir
+    add_tool_to_agent_definition(framework, tool_data, path)  # Add tool to agent definition
+    if tool_data.env: # if the env vars aren't in the .env files, add them
+        # tool_data.env is a dict, key is the env var name, value is the value
+        for var, value in tool_data.env.items():
+            env_var = f'{var}={value}'
+            if not string_in_file(f'{path}.env', env_var):
+                insert_code_after_tag(f'{path}.env', '# Tools', [env_var, ])
+            if not string_in_file(f'{path}.env.example', env_var):
```
Comment (tcdent) [inline agentstack/generation/tool_generation.py:109]: do we need `.env.example`?

Comment (bboynton97) [reply agentstack/generation/tool_generation.py:109]: i personally like keeping it in. if we see confusion from devs that aren't sure what it's meant to be, we can remove support for it. im open to conversation on it though

**Comment by bboynton97** [comment]
it looks like the github workflow is only testing 3.11 and 3.12. ill make a pr to fix this. but 3.9 is still failing with tox when checked out. i'm okay with removing support for 3.9 if we see a good reason

**Review by bboynton97 — APPROVED** [review]

**Comment by bboynton97** [comment]
merged main in, updated `neon.json` to fit the format and added a more descriptive error on parsing failures for testing the tools.

great idea @tcdent on testing all the tool files! i was going to add that in a separate PR then saw you did it here 🙂‍↕️

**Comment by potpiedev** [comment]
## PR Review Started

This PR (qa defaults to multi batch) is now being reviewed. 

Starting Review, should be available shortly

**Pull request merged**